src/main/kotlin/se/uulm/snowballr/backend/repository/UserTableRepo.kt

```diff
@@ -67,13 +87,22 @@ interface IUserTableRepo {
 class UserTableRepo(
     private val db: IDatabase,
 ) : IUserTableRepo {
-    override suspend fun getUserById(id: UUID): User = db.dbQuery {
-        UserTable
+    /**
+     * Requesting a user from the database.
+     *
+     * @param id The id of the requested user.
+     * @return The [User] object or null, if no user with the given [id] was found.
+     */
+    private suspend fun retrieveUserFromDatabase(id: UUID): User? {
+        return UserTable
             .selectAll()
             .where { UserTable.id eq id }
             .map { it.toUser() }
             .singleOrNull()
-            ?: throw NotFoundException.User(id.toString())
+    }
```

Comment:
The "suspend" is not necessary. What about this name? That it's from the database is obvious because we're in the repository layer.
```suggestion
    private fun getUserByIdOrNull(id: UUID): User? =
        UserTable
            .selectAll()
            .where { UserTable.id eq id }
            .map { it.toUser() }
            .singleOrNull()
```

Reply:
Yeh I wanted to differentiate this private function from the `getUserById` function, but probably your wording is better.